LIMS2DB/classes.py

```diff
@@ -203,6 +203,25 @@ def __init__(self, session, log, step):
         self.obj = {}
         self.build()
 
+    def extract_barcode(self, chain):
+        bcp = re.compile("[ATCG\-]{4,}")
+        if "NoIndex" in chain:
+            return chain
+        if '(' not in chain:
+            barcode = chain
+        else:
+            pattern = re.compile("\(([A-Z\-]+)\)")
+            matches = pattern.search(chain)
+            if matches.group(1):
+                barcode = matches.group(1)
+        matches = bcp.match(barcode)
```

Comment:
handled in https://github.com/SciLifeLab/LIMS2DB/pull/48/commits/d2aa88bb2708c5cb0cdf8c0ca403c9247d9d2ff0